### ISSUE TITLE
[Feature Request]: data file stratification / spliting (split files)

### ISSUE THREAD

**Issue by showers14** [jasp-stats/jasp-issues#1379]
* Enhancement:sort cases by AUTHOR definition. 
* Purpose:SPLIT FILE LAYERED BY AUTHOR definition.
* Use-case:

**Is your feature request related to a problem? Please describe.**
 
no
**Describe the solution you'd like**
 
in spss，there is an option called "Split files" in DATA OPTION, author could spilt data into groups, check the results of each group.
**Describe alternatives you've considered**
 
no
**Additional context**

**Comment by vandenman** [comment]
Does this relate to a specific analysis? Perhaps descriptives or meta-analysis?

**Comment by TarandeepKang** [comment]
No, in SPSS this is just a general data management feature. It is applicable across multiple analyses: https://libguides.library.kent.edu/spss/splitdata  relatedly, it would probably also be useful to add a weighting function, this popped into my mind as a similar data management feature that is useful across multiple analyses.

**Comment by vandenman** [comment]
possibly related: https://github.com/jasp-stats/jasp-issues/issues/395 & https://github.com/jasp-stats/jasp-issues/issues/635

**Comment by juliuspf** [comment]
is this feasible, @JorisGoosen ?

**Comment by tomtomme** [comment]
@showers14 
Just to let you know - there is a fix for this now included in the latest nightlies. If you want to test:
https://static.jasp-stats.org/Nightlies/

Be warned that those might be unstable.

See this pull request for how this is supposed to work now via easily removing levels of data:
https://github.com/jasp-stats/jasp-desktop/pull/5789

So you would probably duplicate the variable which you want to use to split the analyis and then remove the levels from that variable which you do not want, and then redo / add copys of the analysis each time with one level. 
This removes the hassle of switching filters on and of or working with different files that are filtered differently.
This is not yet as nice as just having this "split data" button. But it is indeed an improvement.

Please reopen if you disagree that this solves this problem.

**Comment by kornbrot** [comment]
doing it SEPARATELY for every value in column is a pain especially if you are a psycholoist wantting to do separate analyses for each participant. any chance. EASILY????? removing levels of data ! i have > 50 participants. so jasp verioson of https://libguides.library.kent.edu/spss/splitdata is impossible?

**Comment by tomtomme** [comment]
thx for the question.
If I would want to compare sub-groups of a metric variable, I would still use the filter function. I guess you have an ID column for you participant? In the filter window you could e.g. specify "ID < 50" to look only at your first batch of participants. I would save that file. Then I would change the filter to "ID > 50" and save that file under a different name. Now I can do the analyses separately. If I have clicked the analysis in advance, they are already in both files and can be compared side by side.

If I misunderstood your usecase, please share a data screenshot or file. 

In addition: There is work in progress to make the above even faster / easier by using filters per analysis, see: 
https://github.com/jasp-stats/jasp-desktop/pull/5783

this might materialize in final form this year - hopefully

**Comment by tomtomme** [comment]
If you want to split the file via a categorial (nominal ordinal) variable it already is way easier than in earlier times. You would just duplicate your categorial by copy pasting and then remove the not needed levels in each of the two like this:

<img width="1895" height="1985" alt="Image" src="https://github.com/user-attachments/assets/b826d759-cc8e-4a6f-aaa9-15d4f022bb34" />

**Comment by tomtomme** [comment]
this would be the result:

<img width="2390" height="2146" alt="Image" src="https://github.com/user-attachments/assets/f82271ed-39d8-4c6a-96d5-8e74c96d2ef2" />

**Comment by kornbrot** [comment]
i know that. i want to make it run through each vakue of the filter NOT one by one

**Comment by tomtomme** [comment]
Sorry I do not understand. What is "it"? 
And what do you mean with "one by one" vs. "each value". A file with the use case or a video could help here.
